src/commands/data/schemas/util/syncSchema.ts

```diff
@@ -18,41 +16,45 @@ export class SyncSchema {
     this.ds = DataService.createDataService(sdk);
   }
 
-  async sync(target: any) {
-    this.target = target;
+  async sync(target: any, dry?: boolean) {
+    this.localSchema = target;
 
-    if (!this.target.name) {
+    if (!this.localSchema.name) {
       console.log('No schema name defined, skipping this file');
       return;
     }
-    console.log(`Syncing ${this.target.name}`);
 
-    this.current = await this.ds.fetchSchemaByName(this.target.name);
+    console.log(`Syncing ${this.localSchema.name}`);
+    this.cloudSchema = await this.ds.fetchSchemaByName(this.localSchema.name);
 
-    if (!this.current) {
-      this.current = await this.ds.createSchema(this.target.name, this.target.description);
+    if (!this.cloudSchema) {
+      if (dry) {
+        console.log(`\t-> Will be created: ${chalk.green(this.localSchema.name)}`);
+        return;
+      }
+      this.cloudSchema = await this.ds.createSchema(this.localSchema.name, this.localSchema.description);
     }
 
     //  root attributes: update
-    await this.#syncRootAttributes();
+    await this.#syncRootAttributes(dry);
```

Comment:
Seeing at this is class based, can we set dry as a private variable on construction and have it available for the methods.

Seems odd to use it as a parameter in this context.

Reply:
Didn't think of that. Agreed it's a better approach, will update PR